submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/submit/async/MultipartAsyncSubmitResource.kt

```diff
@@ -0,0 +1,136 @@
+package ac.uk.ebi.biostd.submission.web.resources.submit.async
+
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.JSON
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.TSV
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.XML
+import ac.uk.ebi.biostd.submission.converters.BioUser
+import ac.uk.ebi.biostd.submission.domain.service.TempFileGenerator
+import ac.uk.ebi.biostd.submission.web.handlers.SubmitWebHandler
+import ac.uk.ebi.biostd.submission.web.model.ContentSubmitWebRequest
+import ac.uk.ebi.biostd.submission.web.model.FileSubmitWebRequest
+import ac.uk.ebi.biostd.submission.web.model.OnBehalfRequest
+import ebi.ac.uk.extended.model.FileMode
+import ebi.ac.uk.model.constants.ATTRIBUTES
+import ebi.ac.uk.model.constants.FILES
+import ebi.ac.uk.model.constants.FILE_MODE
+import ebi.ac.uk.model.constants.MULTIPART_FORM_DATA
+import ebi.ac.uk.model.constants.SUBMISSION
+import ebi.ac.uk.model.constants.SUBMISSION_TYPE
+import ebi.ac.uk.model.constants.TEXT_PLAIN
+import ebi.ac.uk.model.constants.TEXT_XML
+import ebi.ac.uk.security.integration.model.api.SecurityUser
+import org.springframework.http.HttpHeaders.CONTENT_TYPE
+import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.multipart.MultipartFile
+
+@RestController
+@RequestMapping("/submissions/async")
+@PreAuthorize("isAuthenticated()")
+@Suppress("LongParameterList")
+class MultipartAsyncSubmitResource(
+    private val submitWebHandler: SubmitWebHandler,
+    private val tempFileGenerator: TempFileGenerator
+) {
+    @PostMapping(
```

Comment:
I know we talked about changing the API docs strategy but I think while we define it, we should add it for new controllers